src/global/signal_handler.cc

```diff
@@ -152,6 +155,36 @@ void install_standard_sighandlers(void)
 #include "common/Thread.h"
 #include <errno.h>
 
+void get_name_by_pid(pid_t pid, char *task_name)
+{
+  char proc_pid_path[BUF_SIZE] = {0};
+  snprintf(proc_pid_path, BUF_SIZE, "/proc/%d/cmdline", pid);
+  int file_fd = open(proc_pid_path, O_RDONLY);
+
+  if (file_fd < 0) {
+    derr << "Failure to open '" << proc_pid_path \
+    << "' error= " << file_fd << dendl;
+    return;
+  }
+
+  int ret = read(file_fd, task_name, BUF_SIZE-1);
+  if (ret < 0) {
+    derr << "Failure to read '" << proc_pid_path << "' error= " << ret << dendl;
+  }
+   
+  close(file_fd);
+}
+
+void handle_global_signal_info(int signum, siginfo_t * siginfo, void * content)
+{
+  char task_name[BUF_SIZE] = "unknown";
+  get_name_by_pid(siginfo->si_pid, task_name);
+  derr << "received  signal '" << sys_siglist[signum] \
+  << "' from '" << task_name << "'" << dendl;
+  derr << "'" << task_name << "' PID: " << siginfo->si_pid \
+  << " UID: " << siginfo->si_uid << dendl;
```

Comment:
cosmetic: let's just put it all on one line?  signal a from pid b task name c uid d